src/main/java/uk/ac/ebi/biostudies/auth/CustomRememberMeCookieService.java

```diff
@@ -0,0 +1,166 @@
+                        if(user==null || user.getAllow()==null)
+                            return null;
+                        this.logger.debug("Remember-me cookie accepted");
+                        return this.createSuccessfulAuthentication(httpServletRequest, user);
+                } catch (CookieTheftException exp) {
+                    this.cancelCookie(httpServletRequest, httpServletResponse);
+                    throw exp;
+                } catch (UsernameNotFoundException exp) {
+                    this.logger.debug("Remember-me login was valid but corresponding user not found.", exp);
+                } catch (InvalidCookieException exp) {
+                    this.logger.debug("Invalid remember-me cookie: " + exp.getMessage());
+                } catch (AccountStatusException exp) {
+                    this.logger.debug("Invalid UserDetails: " + exp.getMessage());
+                } catch (Throwable exp) {
+                    this.logger.debug(exp.getMessage());
+                }
+
+                this.cancelCookie(httpServletRequest, httpServletResponse);
+                return null;
+            }
+        }
+    }
+
+    @Override
+    public void loginFail(HttpServletRequest request, HttpServletResponse response) {
+        this.logger.debug("Interactive login attempt was unsuccessful.");
+//        this.cancelCookie(request, response);
+        this.onLoginFail(request, response);
+    }
+
+    protected void onLoginFail(HttpServletRequest request, HttpServletResponse response) {
+    }
+
+    @Override
+    public void loginSuccess(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, Authentication authentication) {
+        if (!this.rememberMeRequested(httpServletRequest, this.parameter)) {
+            this.logger.debug("Remember-me login not requested.");
+        } else {
+            this.onLoginSuccess(httpServletRequest, httpServletResponse, authentication);
+        }
+
+
+    }
+
+    protected boolean rememberMeRequested(HttpServletRequest request, String parameter) {
+        if (this.alwaysRemember) {
+            return true;
+        } else {
+            String paramValue = request.getParameter(parameter);
+            if (paramValue != null && (paramValue.equalsIgnoreCase("true") || paramValue.equalsIgnoreCase("on") || paramValue.equalsIgnoreCase("yes") || paramValue.equals("1"))) {
+                return true;
+            } else {
+                if (this.logger.isDebugEnabled()) {
+                    this.logger.debug("Did not send remember-me cookie (principal did not set parameter '" + parameter + "')");
+                }
+
+                return false;
+            }
+        }
+    }
+
+    public void onLoginSuccess(HttpServletRequest request, HttpServletResponse response, Authentication successfulAuthentication) {
+        String token = ((User)successfulAuthentication.getPrincipal()).getToken();
```

Comment:
Cookies are being set in Authentication::login as well. Should probably move both to a utility method
`Integer maxAge = 31557600;` // should probably be moved to a static constant in Authentication and used there